internal/mode/static/telemetry/collector.go

```diff
@@ -214,3 +219,14 @@ func CollectClusterID(ctx context.Context, k8sClient client.Reader) (string, err
 	}
 	return string(kubeNamespace.GetUID()), nil
 }
+
+// CollectImageSource gets the source of the NGF image. This is done by inspecting the BUILD_AGENT environment variable.
+// Valid sources are: "gha" for images built using GitHub Actions in the pipeline, or "local".
+// If the environment variable is not set to one of these, the source is "unknown"
+func CollectImageSource() string {
+	buildAgent := os.Getenv("BUILD_AGENT")
```

Comment:
Could we localize env variables parsing to one place here https://github.com/nginxinc/nginx-gateway-fabric/blob/92081af949aa6c8436480563659ac8b139135d04/cmd/gateway/commands.go#L118  where  other env vars are already handled?

We can propagate the parsed information in the Config https://github.com/nginxinc/nginx-gateway-fabric/blob/38ebd8c6c7fdac62b1b11c5f855ebeab2e040e36/internal/mode/static/config/config.go#L11